cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedTimeScheduleStore.java

```diff
@@ -365,30 +401,95 @@ public void apply() throws Exception {
       });
   }
 
+  private TriggerKey removeAppVersion(TriggerKey key) {
+    String jobName = key.getName();
+    String[] splits = jobName.split(":");
+    // New TriggerKey name has format = namespace:application:version:type:program:schedule
+    if (splits.length != 6) {
+      LOG.debug("Skip removing app version for trigger key since it doesn't have one : {}", jobName);
+      return null;
+    }
+
+    List<String> splitArray = new ArrayList<>(Arrays.asList(splits));
+    splitArray.remove(2);
```

Comment:
Thanks Mao! Addressed these comments in https://github.com/caskdata/cdap/pull/8486/commits/ef9721399ae903644d9bd5a68ce03f7270ee6239